client/src/pages/NewPost/NewPost.tsx

```diff
@@ -1,51 +1,24 @@
 import React from 'react';
-import { Button } from '@mui/material';
-import { useForm } from '../../hooks/useForm';
-import { Article } from '../../types/initialTypes';
-import { newArticleValidationSchema } from '../../validationSchema';
-import { useStyles, TitleTextField, StoryTextField } from './styles';
+import { useNavigate } from 'react-router-dom';
+
 import { addNewPost } from '../../api/posts';
+import { PostForm } from '../../components/Forms/PostForm';
+import { CreatedPost } from '../../types/initialTypes';
 
 export const NewPost: React.FC = () => {
-  const initialValues = { title: '', content: '' };
-  const { handleChange, handleSubmit, values, errors } = useForm<Article>({
-    validations: newArticleValidationSchema,
-    onSubmit: handleSubmitNewArticle,
-    initialValues,
-  });
-  const classes = useStyles();
+  const navigate = useNavigate();
 
-  const { title, content } = values;
+  const initialValues = { title: '', content: '' };
 
-  async function handleSubmitNewArticle() {
-    const newArticle = { title, content };
-    await addNewPost(newArticle);
-  }
+  const handleSubmitNewArticle = async (post: CreatedPost) => {
+    navigate('/');
+    return await addNewPost(post);
```

Comment:
You do that request in the background basically. Because you redirect first, and then do request. What if something will go wrong with the request, 500 error for example?
It's an interesting UX – to perform operations in the background, but it can lead to some weird behavior. For example, I added a post and I've been redirected to the list of posts, and I don't see that new post, and see it just in a few seconds, for the moment I will probably think that something is broken. Or if I added a post, redirected, and that request failed, I need to start from the very beginning to try again.
Common UX – user stays at the same page, waiting for the request to be finished, and looking at the loader, for example. If everything ok – redirected, if something goes wrong, we don't clean the form or redirect, user can try to send it again.